app/models/manageiq/providers/ibm_cloud/container_manager/metrics_capture/prometheus_capture_context.rb

```diff
@@ -0,0 +1,103 @@
+class ManageIQ::Providers::IbmCloud::ContainerManager::MetricsCapture::PrometheusCaptureContext < ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::PrometheusCaptureContext
+  def initialize(target, start_time, end_time, interval)
+    @target = target
+    @starts = start_time.to_i.in_milliseconds
+    @ends = end_time.to_i.in_milliseconds if end_time
+    @interval = interval.to_i
+    @tenant = target.try(:container_project).try(:name) || '_system'
+    @ext_management_system = @target.ext_management_system
+    @ts_values = Hash.new { |h, k| h[k] = {} }
+    @metrics = []
+
+    @node_hardware = if @target.respond_to?(:hardware)
+                       @target.hardware
+                     else
+                       @target.try(:container_node).try(:hardware)
+                     end
+
+    @node_cores = @node_hardware.try(:cpu_total_cores)
+    @node_memory = @node_hardware.try(:memory_mb)
+
+    validate_target
+  end
```

Comment:
I might just be missing it but does this differ in any way from https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/master/app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context_mixin.rb#L3-L23 which is included in the parent class https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/master/app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/prometheus_capture_context.rb#L4

Reply:
That was my first thought however I found that the mixin was returning a `ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture` object even when being called from https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/f921529b891e2673fcc383675c051cb5006bc91e/app/models/manageiq/providers/ibm_cloud/container_manager/metrics_capture.rb and that led to https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/master/app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context_mixin.rb#L27-L29 calling the parent class methods as opposed to the overridden methods at the IBM Cloud provider level

Reply:
Interesting will have to dig into that one later